src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+
+Note that user defined types are not yet supported in text plans (but will be at a future date).
+
+#### Example Types
+
+|                                        | Explanation                                                         |
+|----------------------------------------|---------------------------------------------------------------------|
+| i64\?                                  | optional int64                                                      |
+| varchar\<5>                            | a string with up to five characters                                 |
+| struct\<string,i8,i32?,timestamp_tz>   | a structure with a string, two ints, one optional, and a timestamp  |
+| struct\<string,struct\<string,string>> | a structure containing a string and a substructure with two strings |
+
+## Literals
+
+Types are important in Substrait plans as the choice of type controls which implementation of a given function should
+be used.  In the text representation of plans we label literals with their types to make it clear which function should
+be invoked within the plan. The type of a literal is appended to the literal behind an underscore (_).  Typed literals 
+may not be optional.
+
+### Integer Literals
+
+There are four different integer widths.  While other languages provide for binary, octal, and hexadecimal
+representations of their integer literals (as well as spacing separators) human readability of literals isn't a current
+requirement of the language and so are omitted for brevity.
+
+#### Example Integer Literals
+```
+123_i8
+0_i16
+123_i32
+1234_i64
+```
+
+### Floating Point Literals
+
+Floating point literals are represented with by a decimal number followed by an optional exponent.   
+
+#### Example Floating Point Literals
+
+```
+-1_fp32
+0.3_fp64
+1.99E-13_fp64
+```
+
+### Decimal Literals
+
+Decimal literals are similar to that of floating point literals however are represented by a real number with an
+exponent which is specified as part of the type.  The type includes the precision and the scale (number of base 10
+digits to translate the result by). 
+
+#### Example Decimal Literals
+
+```
+-123_decimal<3,0>
+42_decimal<5,-4>
+```
+
+### String Literals
+
+No need for multiline.  Raw strings may be useful for regexes.  Strings with no size limit may be specified without a
+type specifier for brevity.
+
+| Escape Sequence | Description                         |
+|-----------------|-------------------------------------|
+| \n              | linefeed (ASCII code 10)            |
+| \r              | carriage return (ASCII code 13)     |
+| \t              | tab (ASCII code 8)                  |
+| \\              | backslash                           |
+| \'              | single quote                        |
+| \"              | double quote                        |
+| \xNN            | two byte hexadecimal character code |
+| \u{NNNNNNN}     | unicode code sequence               |
```

Comment:
Is this minimal set of escape sequences based on an existing implementation?  E.g. I think this looks a bit like Java.  Do we have confidence it is sufficiently complete?